### ISSUE TITLE
floating_ip_assignment with create_before_destroy

### ISSUE THREAD

**Issue by dave-lew99** [terraform-providers/terraform-provider-openstack#439]
Hi there,

Terraform Version
Terraform v0.11.8

Affected Resource(s)
Please list the resources as a list, for example:

openstack_compute_instance_v2
openstack_compute_floatingip_associate_v2
If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

Terraform Configuration Files
```resource "openstack_compute_instance_v2" "test-server" {
    name            = "test-tf"
    flavor_id       = "003386be-0c35-427c-a0cc-0789ebb8f79b" # this is the instance type `openstack flavor list`

    block_device {
        uuid                  = "f61601e5-c9e1-4def-a107-bb082f914a61" # this is the image ID to boot from `openstack image list`
        source_type           = "image"
        volume_size           = 5
        boot_index            = 0
        destination_type      = "volume"
        delete_on_termination = true
    }

    network {
    name = "Public"
    }

    lifecycle {
        create_before_destroy = true
    }
}

resource "openstack_compute_floatingip_associate_v2" "myip" {
    floating_ip = "<public ip>"
    instance_id = "${openstack_compute_instance_v2.test-server.id}"
    lifecycle {
        create_before_destroy = true
    }
}
```

Debug Output
Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the crash.log.

Expected Behavior
When updating an openstack instance with a new image, the new instance should be provisioned then the floating the IP assignment is changed over to the new instance, before the old instance is destroyed.

Actual Behavior
The new instance is created and the floating IP is swapped over to the new instance. The apply then fails with the message 

> openstack_compute_floatingip_associate_v2.myip (deposed #0): floating ip association: Expected HTTP response code [201 202] when accessing [POST https://<openstack url>/compute/v2.1/servers/d93cb36e-fdeb-44d3-9068-0e73e2200dc9/action], but got 409 instead
{"conflictingRequest": {"message": "Floating IP <ip> is not associated with instance d93cb36e-fdeb-44d3-9068-0e73e2200dc9.", "code": 409}}

Steps to Reproduce
Please list the steps required to reproduce the issue, for example:

terraform apply
Change the image value
Run terraform apply
References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

Maybe https://github.com/terraform-providers/terraform-provider-digitalocean/issues/147

**Comment by ozerovandrei** [comment]
Hello.

Resource `openstack_compute_floatingip_associate_v2` uses old Compute API handler for managing floating IP associations, that API is not available starting from the [2.44](https://developer.openstack.org/api-ref/compute/#add-associate-floating-ip-addfloatingip-action-deprecated) OpenStack Compute service microversion.

It's better to use Networking API to manage floating IPs associations with the [openstack_networking_floatingip_associate_v2](https://www.terraform.io/docs/providers/openstack/r/networking_floatingip_associate_v2.html) resource.

You can try the following example:

```hcl
data "openstack_networking_network_v2" "network_1" {
  name = "Public"
}

resource "openstack_networking_port_v2" "port_1" {
  network_id = "${data.openstack_networking_network_v2.network_1.id}"
}

resource "openstack_compute_instance_v2" "instance_1" {
  name = "test-tf"
  flavor_id = "003386be-0c35-427c-a0cc-0789ebb8f79b"
  
  network {
    port = "${openstack_networking_port_v2.port_1.id}"
  }

  block_device {
    uuid = "f61601e5-c9e1-4def-a107-bb082f914a61"
    source_type = "image"
    volume_size = 5
    boot_index = 0
    destination_type = "volume"
    delete_on_termination = true
  }

  lifecycle {
    create_before_destroy = true
  }
}
```

If you change `block_device.uuid`, instance will be recreated with the same port and associated floating IP.

**Comment by dave-lew99** [comment]
Thanks for the help - i've converted over to using the networking APIs/ports etc, but i'm still having trouble with the create_before_destroy of instances - it gives me an error that the port is still in use when trying to launch the replacement instance.

`{"conflictingRequest": {"message": "Port a20ab8a3-f198-4db6-9623-e21dacff1d98 is still in use.", "code": 409}}`

The minimal code required to run up an instance is here - 

```ruby
resource "openstack_networking_router_v2" "internet-gateway" {
  name                = "internet-gateway"
  admin_state_up      = true
  external_network_id = "ab2264dd-bde8-4a97-b0da-5fea63191019" # from openstack network list - provided by selectel
}

resource "openstack_networking_router_interface_v2" "internet-gateway-interface" {
  router_id = "${openstack_networking_router_v2.internet-gateway.id}"
  subnet_id = "${openstack_networking_subnet_v2.public-subnet.id}"
}

resource "openstack_networking_network_v2" "public-network" {
  name           = "public-network"
  admin_state_up = "true"
}

resource "openstack_networking_subnet_v2" "public-subnet" {
  name       = "public-subnet"
  network_id = "${openstack_networking_network_v2.public-network.id}"
  cidr       = "192.168.1.0/24"
  ip_version = 4
}

resource "openstack_networking_floatingip_v2" "floating-ip-1" {
  pool = "external-network"
}

resource "openstack_networking_port_v2" "net_port_1" {
  network_id = "${openstack_networking_network_v2.public-network.id}"
  fixed_ip {
    subnet_id  = "${openstack_networking_subnet_v2.public-subnet.id}"
    ip_address = "192.168.1.50"
  }
  admin_state_up = "true"
}

resource "openstack_compute_instance_v2" "instance-1" {
    name            = "instance-1"
    flavor_id       = "003386be-0c35-427c-a0cc-0789ebb8f79b" # this is the instance type openstack flavor list

block_device {
        uuid                  = "a73de7bb-cdee-49f9-a725-e045769b2470" # this is the image ID to boot from openstack image list
        source_type           = "image"
        volume_size           = 5
        boot_index            = 0
        destination_type      = "volume"
        delete_on_termination = true
    }

    network {
        port = "${openstack_networking_port_v2.net_port_1.id}"
    }

    lifecycle {
        create_before_destroy = true
    }
}

resource "openstack_networking_floatingip_associate_v2" "instance-1-ip-assoc" {
    floating_ip = "${openstack_networking_floatingip_v2.floating-ip-1.address}"
    port_id = "${openstack_networking_port_v2.net_port_1.id}"

}
```

Removing the create_before_destroy on the instance allows instance replacement to work (and keep the same floating IP) but results in a temporary outage between the old instance being destroyed and the new instance being fully booted - which is what i am trying to avoid.

**Comment by ainsey11** [comment]
I too am having this issue

**Comment by rnurgaliyev** [comment]
Having exactly same issue, downtime because of early floating up reassignment. Did you find any solution?

**Comment by jtopjian** [comment]
This has been fixed in #761.
